R/variables.R

```diff
@@ -0,0 +1,87 @@
+#' @title Continuous age variable
+#' @description Create a continuous age variable for the population
+#'
+#' @param pop population list
+#' @param max_age maximum age to be drawn
+#'
+#' @return continuous age variable
+#' @importFrom stats dexp
+create_continuous_age_variable <- function(pop, max_age = 100) {
+
+  # get out country median ages
+  iso3c <- pop$iso3c[1]
+  iso3c_ages <- hypatia::iso3c_ages
+  med_age <- iso3c_ages$age[iso3c_ages$iso3c == iso3c]
+
+  # get the top end of the 5 year age bins
+  age_bins <- c(0, as.numeric(gsub("^(\\d{1,2}).*", "\\1", pop$age_group)[-1]))
+
+  # use these to work out the ages in each bin
+  r <- list()
+  for (i in seq_along(age_bins)[-1]) {
+    r[[i - 1]] <- seq(age_bins[i - 1], age_bins[i] - 1, 1)
+  }
+  r[[length(r) + 1]] <- seq(max(age_bins), max_age, 1)
+
+  # now sample from these
+  ages <- list()
+  for (i in seq_len(length(pop$age_group))) {
+    ages[[i]] <- sample(r[[i]], pop$n[i], replace = TRUE,
```

Comment:
The reason why your tests are failing is that `pop$n` is not an integer here, but is a floating point number that *looks* like an integer and has suffered some rounding error at some point.

```
> floor(pop$n) - round(pop$n)
 [1]  0  0  0  0  0  0  0  0  0  0 -1  0  0  0  0  0  0
```

`sample` is doing an integer cast on the number which is where your -1 is coming from in the test (which shows the test is working!).

The solution is to round the numbers in your data processing script, I think

Reply:
ok, was tired last night when committing so couldn't see the wood for the trees. Will look now

Reply:
This does stem from `squire` not rounding model estimates of the demography. This was fixed in squire `v0.5.5` (https://github.com/mrc-ide/squire/blob/master/tests/testthat/test-population.R#L108-L113) so change the dependency in DESCRIPTION as well.